opentech/public/utils/models.py

```diff
@@ -259,3 +260,39 @@ class Meta:
         SocialFields.promote_panels +
         ListingFields.promote_panels
     )
+
+
+class BaseFunding(Orderable):
+    value = models.PositiveIntegerField()
+    year = models.PositiveIntegerField()
+    duration = models.PositiveIntegerField(help_text='In months')
+    source = models.ForeignKey(
+        'wagtailcore.Page',
+        on_delete=models.PROTECT,
+    )
+
+    panels = [
+        FieldRowPanel([
+            FieldPanel('year'),
+            FieldPanel('value'),
+            FieldPanel('duration'),
+        ]),
+        PageChooserPanel('source', ['public_funds.FundPage', 'public_funds.LabPage']),
+    ]
+
+    class Meta(Orderable.Meta):
+        abstract = True
+
+
+class FundingMixin(models.Model):
+    '''Impliments the funding total calulation
+
+    You still need to include the InlinePanel('funding') in the child class
```

Comment:
Is there no way of making this so one just needs to do `panels + FundingMixin.panels` ?

Reply:
That is the correct pattern. Thank you.